ocaml/xapi/import.ml

```diff
@@ -275,14 +275,26 @@ let assert_can_live_import __context rpc session_id vm_record =
            )
         )
   in
+  let dmc_compatible =
+    Xapi_vm_helpers.is_dmc_compatible_vmr ~__context ~vmr:vm_record
+  in
   match vm_record.API.vM_power_state with
-  | `Running ->
+  | `Running when dmc_compatible ->
       assert_memory_available ()
-  | `Paused ->
+  | `Paused when dmc_compatible ->
       assert_memory_available ()
-  | `Suspended ->
-      () (* disable DMC *)
-  | _other ->
+  | `Suspended when dmc_compatible ->
+      ()
+  | _ when not dmc_compatible ->
```

Comment:
May be easier to just do `if Xapi_vm_helpers.is_dmc_compatible_vmr ~__context ~vmr:vm_record then ...log and raise error...` above the match to avoid all the `when` clauses.

Reply:
Not also that this function is only called for cross-pool live migration. We need to also handle the normal upgrade case.

Reply:
I think a flat `match` structure is easier to read and reason about  compared to nested `if` and it avoids the problem of multiple statements in `if` that need grouping. But it's a matter of taste.

Reply:
I think I will try to unify the handling of live and other imports - take back some of the changes I made.